src/cmd/contact/delete.sh

```diff
@@ -23,10 +23,13 @@ cmd_contact_delete() {
     [[ -z $1 ]] && fail "Usage:\n$(cmd_contact_delete_help)"
 
     if [[ $force == 0 ]]; then
-        gpg --delete-keys "$@"
-    else
-        gpg --batch --no-tty --yes --delete-keys "$@"
+        yesno "Delete contact(s). Are you sure?" || return 0
```

Comment:
This is does not preserve the original behavior.
The question itself is meaningless and does not add any extra protection, without showing the contacts that are going to be deleted.